brushfire-scalding/src/main/scala/com/stripe/brushfire/scalding/Trainer.scala

```diff
@@ -185,6 +185,40 @@ case class Trainer[K: Ordering, V, T: Monoid](
   }
 
   /**
+   * prune a tree to minimize validation error
+   *
+   * Construct a Map[Int,T] from the trainingData for each tree, and then transform the trees using the prune method.
+   *
+   * writeThrough(TreeSource(path))
+   */
+  def prune[P, E](path: String, error: Error[T, P, E])(implicit voter: Voter[T, P], inj: Injection[Tree[K, V, T], String], ord: Ordering[E]): Trainer[K, V, T] = {
+    flatMapTrees {
+      case (trainingData, sampler, trees) =>
+        lazy val treeMap = trees.toMap
+
+        val newEx: Execution[TypedPipe[(Int, Tree[K, V, T])]] = trainingData
+          .flatMap { instance =>
+            for { // Iterate over any trees for which this instance is a validation instance.
+              (treeIndex, tree) <- treeMap if sampler.includeInValidationSet(instance.id, instance.timestamp, treeIndex);
+              leafIndex <- tree.leafIndexFor(instance.features).toList // Find the leaf that this instance falls into.
+            } yield (treeIndex, leafIndex) -> instance.target
+          } // Associate the tree and leaf indices with the instance target.
+          .sumByKey // Combine validation data for each leaf in each tree. Q: Why is this necessary?
```

Comment:
I don't think it is necessary. I think you could do `yield treeIndex -> Map(leafIndex -> target)` above and go right to the group/sum below. I'm guessing this is cargo culted off something else which maybe has the same problem?